example/box_example.cpp

```diff
@@ -87,12 +87,12 @@ int main()
 
   constexpr auto mm = isq::length[unit_symbols::mm];  // helper reference object
 
-  const auto height = (200.0 * mm)[metre];
-  auto box = Box(1000.0 * mm, 500.0 * mm, height);
-  box.set_contents_density(1000.0 * isq::mass_density[kg / m3]);
+  const auto height = mm(200.0)[metre];
+  auto box = Box(mm(1000.0), mm(500.0), height);
+  box.set_contents_density(isq::mass_density(1000.0, kg / m3));
```

Comment:
This doesn't seem like an improvement to me.

Reply:
Same for me, but as we stated, there will be users that would like to provide a short name for reference (maybe even you 😉). This is how it will look in such cases. Do you have any ideas on how to improve that? I would like to leave the multiply syntax for just this corner case usage. Besides physical constants, the library does not provide any named references.

Reply:
`mm` is a misnomer. It's actually a quantity and a unit without a number.

Reply:
`mm` above is exactly the same as `mm` on the mainline now:
https://github.com/mpusz/units/blob/02d6e104f09f47a22eb5994974e5c7957becada2/src/systems/si/include/units/isq/si/length.h#L309-L310

 If I recall correctly, you introduced that feature a long time ago 😉

Reply:
I think those are your unit aliases.

Reply:
Right, my fault. Fixed above.

Reply:
I think I suggested `reference`. I'm having trouble finding the history behind it. In V2, it seems that `reference` takes a quantity rather than a dimension.

I retract https://github.com/mpusz/units/pull/391#discussion_r1055733538. It's actually right, but a bit confounding to have 2 identifiers `mm`. One is a unit, that when multiplied by a number is not enough to specify a quantity. The other is a reference, that by definition, its multiplication by a number is a quantity. Although it's a bit contradictory that the definition of quantity says that the reference can be a unit of measurement.

Reply:
`reference` does not take a `quantity` but a `quantity_spec` which is basically a `dimension` on steroids.

Please note that user-specific shortcuts will be in different project-specific namespaces, so they will not collide with unit symbols in mp-units.  Their names will be probably similar to `m`, `mm`, `km`, etc to simplify common quantities constructions. Also, most probably, such a user will not even use the `unit_symbols` namespace at all but will just use long unit names to not have symbols collisions.

Reply:
Yeah, that's why I didn't use code font.